src/back/glsl/mod.rs

```diff
@@ -2252,36 +2252,51 @@ impl<'a, W: Write> Writer<'a, W> {
                             }
                             _ => write!(self.out, "{}", scalar.full)?,
                         }
+
+                        write!(self.out, "(")?;
+                        self.write_expr(expr, ctx)?;
+                        write!(self.out, ")")?
                     }
                     None => {
                         use crate::ScalarKind as Sk;
 
                         let source_kind = inner.scalar_kind().unwrap();
-                        write!(
-                            self.out,
-                            "{}",
-                            match (source_kind, target_kind) {
-                                (Sk::Float, Sk::Sint) => "floatBitsToInt",
-                                (Sk::Float, Sk::Uint) => "floatBitsToUInt",
-                                (Sk::Sint, Sk::Float) => "intBitsToFloat",
-                                (Sk::Uint, Sk::Float) => "uintBitsToFloat",
-                                // There is no way to bitcast between Uint/Sint in glsl. Use constructor conversion
-                                (Sk::Uint, Sk::Sint) => "int",
-                                (Sk::Sint, Sk::Uint) => "uint",
-                                _ => {
-                                    return Err(Error::Custom(format!(
-                                        "Cannot bitcast {:?} to {:?}",
-                                        source_kind, target_kind
-                                    )));
-                                }
+                        let (conv_op, need_parentheses) = match (source_kind, target_kind) {
```

Comment:
the `need_parentheses` seems redundant, we could just check `conv_op.is_empty()`